### PR TITLE
Switch UI build system to pnpm

### LINKED ISSUE #66: Document the use of npm in `ui`
Mention `npm run`, etc.

### PR DESCRIPTION
**Pull request by MindTooth**
I've changed the setup to [`pnpm`](https://pnpm.io) as more and more projects are doing the same.  I've noted quite an increase in speed.

Also includes some other tweaks:

* fixed legacy Tailwind option
* inlined all resources
* bump prettier
* tweaked the documentation a bit (this needs a thorough rewrite)
* moved types to regular deps (same package in same group)

## Instruction for publishing

```shell
cd src/ui
pnpm install
pnpm run build:prod
```

Fixed #66.

### CODE REVIEW

File: .github/workflows/test-ui.yaml [hunk 1]
```diff
@@ -10,18 +10,22 @@ on:
...
+        uses: ./.github/actions/build-ui
+
       - uses: actions/setup-python@v4
         with:
           python-version: "3.10"
-      - uses: ./.github/actions/build-ui
+          cache: "pip"
       - run: |
           python -m pip install --upgrade pip setuptools wheel
           pip install --upgrade tox virtualenv
```
Comment (MindTooth) [inline .github/workflows/test-ui.yaml]: Could we clean this a bit for all actions and instructions?

Ref. action, could we clean a bit?:
```yaml
steps:
- uses: actions/checkout@v3
- uses: actions/setup-python@v4
  with:
    python-version: '3.9'
    cache: 'pip' # caching pip dependencies
- run: pip install -r requirements.txt
```

to something like:

Locally I could do with just for building:

```shell
pip install --upgrade pip build setuptools
python3 -m build
```

And for testing

```shell
pip install --upgrade pip tox
tox -e py39
```

`tox` installs all it needs into an isolated environment anyways.

@tomrtk thoughts?  I've done the cleanest steps myself above and it works.  Yet to try them on Windows that is.

Comment (tomrtk) [reply .github/workflows/test-ui.yaml]: Sure, try it:)

File: src/ui/README.md [hunk 2]
```diff
@@ -5,43 +5,18 @@ Flask. For the frontend we use TailwindCSS and jQuery/jsTree.
...
-Run `npm run` to see available tasks to run when developing the frontend.
-`npm run build:dev` is used when a minified version of the javascript or css is not
-desirbed. Use `npm run build:prod` otherwise. **Note!** It's best to make sure
-that you always minify before pushing upstream. The reasons that currently the
-project ship both `src` and `dist`.
-
-#### Compile changes to CSS/JavaScript
-
-Using the command below, but are compiled at once. View `npm run` for other options.
+Using the command below, but are compiled at once. View `pnpm run` for other options.
```
Comment (tomrtk) [inline src/ui/README.md]:
```suggestion
Use the command below. View `pnpm run` for other options.
```

File: README.md [hunk 3]
```diff
@@ -13,21 +13,22 @@ computer scientists at the Norwegian University of Science and Technology
...
-  [`virtualenv`](https://virtualenv.pypa.io) (recommended)
-- The URL for downloading the package
+It is recommended to install the application in a virtual environment using for
+example [virtualenv](https://virtualenv.pypa.io) or
+[`venv`](https://docs.python.org/3/library/venv.html).
 
 Find url to latest `whl`
-[here](https://github.com/MindTooth/fish-code/releases/latest) for use in
-the commands below.
+[here](https://github.com/MindTooth/fish-code/releases/latest) for use in the
```
Comment (tomrtk) [inline README.md]:
```suggestion
[here](https://github.com/tomrtk/fish-code/releases/latest) for use in the
```

File: .github/workflows/test-ui.yaml (hunk 1 above)
Comment (MindTooth) [reply .github/workflows/test-ui.yaml]: Seems to work. 👍🏻

**Comment by MindTooth** [comment]
I think it's done now.

I'll work on the release pipeline next.  But, first I'll see if I can tweak the running of Flask with watch or something.  So it's easier to developer at the same time both Python and CSS/JS code in development mode.

**Review by tomrtk — APPROVED** [review]
Ok, will you squeeze the commits? as some of them look to not be passing ci if checked out?

**Pull request merged**